src/java.desktop/share/classes/javax/swing/plaf/basic/BasicSplitPaneDivider.java

```diff
@@ -360,6 +360,20 @@ else if (e.getPropertyName() == JSplitPane.
         }
     }
 
+    /**
+     * {@inheritDoc}
+     * @param enabled {@inheritDoc}
+     */
+    @Override
+    public void setEnabled(boolean enabled) {
+        if (splitPane.isOneTouchExpandable() &&
```

Comment:
i think for setEnabled the buttons should be enabled/disabled regardless of if splitPane isOneTouchExpandable is true so that the state is preserved even if setEnabled(false) is called before 
setOneTouchExpandable(true)

Reply:
No, the left and right arrow buttons are created ONLY when `isOneTouchExpandable` is true so it should be checked here too to enable/disable actions

Reply:
Alisen's concern is valid.

What if `setEnabled(false)` is called when `isOneTouchExpandable` is `false` and then `setOneTouchExpandable(true)` is called which adds the buttons? The buttons are enabled when they should be disabled.

Reply:
Why buttons will be enabled? The buttons state is determined by `setEnabled `param, if `setEnabled `is false, then irrespective of `isOneTouchExpandable`, it will be disabled...`isOneTouchExpandable `is just extra layer of check as that determines whether arrow buttons are rendered or not....If oneTouchExpandable is not there then arrow buttons itself is not created so no point set their state...